[2024-07-27T20:43:58.995+0100] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-27T20:43:59.024+0100] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: jobs_etl.glassdoor_scraper scheduled__2024-07-26T18:00:00+00:00 [queued]>
[2024-07-27T20:43:59.030+0100] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: jobs_etl.glassdoor_scraper scheduled__2024-07-26T18:00:00+00:00 [queued]>
[2024-07-27T20:43:59.030+0100] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-07-27T20:43:59.039+0100] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): glassdoor_scraper> on 2024-07-26 18:00:00+00:00
[2024-07-27T20:43:59.048+0100] {standard_task_runner.py:63} INFO - Started process 47261 to run task
[2024-07-27T20:43:59.052+0100] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'jobs_etl', 'glassdoor_scraper', 'scheduled__2024-07-26T18:00:00+00:00', '--job-id', '229', '--raw', '--subdir', 'DAGS_FOLDER/jobs_etl.py', '--cfg-path', '/tmp/tmpii4_vge8']
[2024-07-27T20:43:59.054+0100] {standard_task_runner.py:91} INFO - Job 229: Subtask glassdoor_scraper
[2024-07-27T20:43:59.120+0100] {task_command.py:426} INFO - Running <TaskInstance: jobs_etl.glassdoor_scraper scheduled__2024-07-26T18:00:00+00:00 [running]> on host huser
[2024-07-27T20:43:59.216+0100] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='hassounibarka@gmail.com' AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='jobs_etl' AIRFLOW_CTX_TASK_ID='glassdoor_scraper' AIRFLOW_CTX_EXECUTION_DATE='2024-07-26T18:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-26T18:00:00+00:00'
[2024-07-27T20:43:59.217+0100] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-27T20:44:29.073+0100] {logging_mixin.py:188} INFO - 15 jobs found for data Scientist
[2024-07-27T20:47:59.807+0100] {logging_mixin.py:188} INFO - 1 Jobs extracted2 Jobs extracted3 Jobs extracted4 Jobs extracted5 Jobs extracted6 Jobs extracted7 Jobs extracted8 Jobs extracted9 Jobs extracted10 Jobs extracted11 Jobs extracted12 Jobs extracted13 Jobs extracted14 Jobs extracted15 Jobs extracted16 jobs found for data analyst
[2024-07-27T21:00:21.399+0100] {logging_mixin.py:188} INFO - 1 Jobs extracted2 Jobs extracted3 Jobs extracted4 Jobs extracted5 Jobs extracted6 Jobs extracted7 Jobs extracted8 Jobs extracted9 Jobs extracted10 Jobs extracted11 Jobs extracted12 Jobs extracted13 Jobs extracted14 Jobs extracted15 Jobs extracted16 Jobs extractedKeyword error
[2024-07-27T21:00:21.477+0100] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-27T21:00:21.477+0100] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/huser/airflow_workspace/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/huser/airflow_workspace/airflow_env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/huser/airflow_workspace/airflow_env/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
  File "/home/huser/airflow_workspace/airflow_env/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/huser/airflow_workspace/airflow_env/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/huser/airflow_workspace/airflow/dags/jobs_etl.py", line 31, in get_glassdoor_data
    df = glassdoor_scraper.get_datas(Keywords)
  File "/home/huser/airflow_workspace/airflow/dags/Jobs/GlassdoorScraper.py", line 143, in get_datas
    links = get_links(key)
  File "/home/huser/airflow_workspace/airflow/dags/Jobs/GlassdoorScraper.py", line 69, in get_links
    return Links
UnboundLocalError: local variable 'Links' referenced before assignment
[2024-07-27T21:00:21.496+0100] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=jobs_etl, task_id=glassdoor_scraper, run_id=scheduled__2024-07-26T18:00:00+00:00, execution_date=20240726T180000, start_date=20240727T194359, end_date=20240727T200021
[2024-07-27T21:00:21.513+0100] {standard_task_runner.py:110} ERROR - Failed to execute job 229 for task glassdoor_scraper (local variable 'Links' referenced before assignment; 47261)
[2024-07-27T21:00:21.557+0100] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-27T21:00:21.588+0100] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-27T21:00:21.589+0100] {local_task_job_runner.py:222} INFO - ::endgroup::
